php/class-wp-cli.php

```diff
@@ -45,6 +45,120 @@ public function create_guest_authors( $args, $assoc_args ) {
 		WP_CLI::line( "- {$skipped} users already had guest author profiles" );
 	}
 
+	/**
+	 * Delete guest authors.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <guest-author-ids-file>
+	 * : Provide the location of a CSV file containing a list of two columns: guest author ID to delete, and either the string "false" (which means the deleted guest author's posts will not be reassigned) or a string denoting the login of the user or guest author the deleted guest author's posts are to be reassigned to. No column headers are required or allowed, and the maximum line length including ID, login, and comma is 100 characters.
+	 *
+	 * [--dry-run]
+	 * : By default, the command outputs the IDs of the guest authors it would have attempted to delete, had this not been a dry run. When you are ready to modify data, pass --no-dry-run.
+	 *
+	 * ## EXAMPLES
+	 *
+	 * # Dry run version of deleting guest authors.
+	 * $ wp co-authors-plus delete-guest-authors testFile.csv
+	 * DRY RUN
+	 * Attempting to delete guest author with ID 1, intended assignee smistephen
+	 *
+	 * # Actually deleting guest authors.
+	 * $ wp co-authors-plus delete-guest-authors testFile.csv --no-dry-run
+	 * You have provided the --no-dry-run flag. Please confirm you wish to delete guest authors. [y/n]
+	 * Successfully deleted guest author 1
+	 * OR
+	 * Error while attempting to delete guest author 1: Guest author does not exist.
+	 *
+	 * @subcommand delete-guest-authors
+	 * @synopsis <guest-author-ids-file> [--dry-run]
+	 */
+	public function delete_guest_authors( $args, $assoc_args ) {
+		global $coauthors_plus;
+
+		$defaults = array(
+			'dry-run' => true,
+		);
+
+		$assoc_args = wp_parse_args( $assoc_args, $defaults );
+
+		$dry_run = $assoc_args['dry-run'];
+
+		// Safety first.
+		$filename_status = validate_file( $args[0] );
+
+		if ( 0 !== $filename_status ) {
+			WP_CLI::error( __( 'The provided file cannot be accessed. Please use either relative or absolute pathnames, and avoid using directory traversal such as ../ and other methods.', 'co-authors-plus' ) );
+		}
+
+		$filename = $args[0];
+
+		// Add custom error handling on fopen error.
```

Comment:
I felt it was a clearer variable name.

Reply:
Sorry, I meant the custom error handling, not the variable name :)

Reply:
Ah. The custom error handling is there to avoid an ugly PHP warning and stack trace from being thrown to the console if the CSV file can't be opened, doesn't exist, or some other error.